### ISSUE TITLE
Rules around the “incode” bit of the org ID

### ISSUE THREAD

**Issue by andylolz** [OpenDataServices/org-ids#195]
I have a question [related to this comment](https://github.com/ViderumGlobal/ckanext-iati/issues/128#issuecomment-366303215).

Am I correct in thinking `"BE-BCE_KBO-0421210424"`, and `"BE-BCE_KBO-0421210424 "` (i.e. with a space on the end) are two separate and equally valid org IDs?

Or rather: that it’s down to the business rules of the `BE-BCE_KBO` list to decide otherwise?

Relatedly: Does [org-id.guide](http://org-id.guide) impose any rules about what constitutes a valid / invalid agency code? If not, (and this is the main question, really) should the related [IATI standard ruleset rule](https://github.com/IATI/IATI-Rulesets/blob/e77e6860ceba3be71d95c7ed95b9c78f3f29928e/rulesets/standard.json#L33) be relaxed/removed?

**Comment by BobHarper1** [comment]
I would ask, why the space? If the space isn't part of the identifier that the registration agency (BCE) gives out, so I wouldn't expect to see it form part of a valid code (what's to stop any number of whitespaces being used?)

There is no true consistency around formatting of agency codes. It is usually the "how to located and use identifiers" section that gives the guidance around what to do. But in most cases if there is whitespace within the agency code itself I'm _assuming_ that the guidance would be to remove these ([see also](http://docs.org-id.guide/en/latest/metadata/#how-to-locate-identifiers)). 

I've found that agency website might use the codes in two different ways themselves in their own interfaces (e.g. with or without non-alphanumeric characters), so the guidance part is important as to removing ambiguity - but not always consistent from one list to the other.

So we could actually say that `"BE-BCE_KBO-0421.210.424"` (as it is on the [search site](http://kbopub.economie.fgov.be/kbopub/zoeknummerform.html;jsessionid=AEA78D7B8B8266D61589799E0E656584.worker4a?nummer=0421210424&actionLu=Search)) is actually the valid org-id because [BE-BCE_KBO](http://org-id.guide/list/BE-BCE_KBO) doesn't say to remove dots.

**Comment by andylolz** [comment]
> There is no true consistency around formatting of agency codes. It is usually the "how to locate and use identifiers" section that gives the guidance around what to do

Exactly – that’s what I meant by “it’s down to the business rules of the BE-BCE_KBO list to decide otherwise”. Okay, great.

> I'm _assuming_ that the guidance would be to remove these

Well, the guidance doesn’t say anything about whitespace… So it would be wrong to assume this, wouldn’t it? This is the core of my question, really – i.e. is there any stuff that [org-id.guide](http://org-id.guide) assumes, that isn’t part of the registration agency’s own rules? Perhaps in this case whitespace isn’t important, but for other lists it might be – so I’d guess it’s probably best not to assume.

The point of my question is: IATI has some fairly arbitrary rules about org IDs (question marks, pipes and backslashes are not allowed), and I wonder if these should cease to exist now that the [org-id.guide](http://org-id.guide) approach is to be followed. This is particularly relevant, because Viderum are talking about introducing validation on org IDs in the IATI registry. They could check for question marks, pipes and backslashes, but I think they probably shouldn’t.

**Comment by BobHarper1** [comment]
> Well, the guidance doesn’t say anything about whitespace… So it would be wrong to assume this, wouldn’t it?

Yes, but it there's no whitespace in the original agency code in this example, why would it be introduced? (and how would we detail its removal if there were none to remove?) But, in any case, what I'm saying is in most cases the business rules would say to remove whitespace... _I assume_!

> The origin of my question is: IATI has some fairly arbitrary rules about org IDs (forward slashes, ampersands, pipes and question marks are not allowed), and I wonder if these should cease to exist now that the org-id.guide approach is to be followed

I would agree, in principle. The IATI Standard could, in theory, apply its own rules and ignore those on the site to support its own use case, but that would seem to be at cross purposes.

**Comment by andylolz** [comment]
> in most cases the business rules would say to remove whitespace

I think we’re agreeing here. I’m asking about the rules [org-id.guide](http://org-id.guide) introduces – not the rules of the registration agency. I think you’re saying: there aren’t any. The only rules are those of the registration agency. If whitespace is not allowed, that’s the registration agency’s call (and NOT org-id.guide.)

That said… Looking at [UG-RSB](http://org-id.guide/list/UG-RSB), for instance… It looks like spaces in the examples have been stripped out ([source](https://github.com/org-id/register/blob/309617d5c3037e517388d7785f58e9b5f25fae15/lists/ug/ug-rsb.json#L46)). So perhaps this is a rule of [org-id.guide](http://org-id.guide) after all?

On the second part of the question, consider `ZW-PVO-42/10`. This is a totally valid org ID according to [the guidelines on org-id.guide](http://org-id.guide/list/ZW-PVO) (indeed, it’s one of the examples provided). However, it’s invalid according to current IATI rules, because it includes forward slashes. That’s problematic if IATI is following the [org-id.guide](http://org-id.guide) methodology.

**Comment by andylolz** [comment]
[The whitespace stripping in this example is crazy bananas](http://org-id.guide/list/PT-NIPPC):

> PT-NIPPC-9digits;thelastdigitisthecheckdigit.Thefirstdigitdependsonwhatthenumberrefersto(5arecompanies.)

**Comment by BobHarper1** [comment]
Whoa, thanks. Looks like a how to find... value ended up in the example property by mistake

**Comment by andylolz** [comment]
Yeah there are a few broken examples (e.g. [GG-RCE](http://org-id.guide/list/GG-RCE); the first example on [FR-RCS](http://org-id.guide/list/FR-RCS)) but my point is: whitespace is being stripped from all examples that contain it.

**Comment by andylolz** [comment]
We basically didn’t get to the bottom of this.
